openid-4-verifiable-presentations-1_0.md

```diff
@@ -616,11 +622,310 @@ The Wallet then validates the request as specified in OAuth 2.0 [@RFC6749].
+  the first option that it can satisfy. If the Wallet cannot satisfy any of the
+  options, it MUST NOT return any claims.
+
+When a Claims Query contains a restriction on the values of a claim, the Wallet
+SHOULD NOT return the claim if its value does not match at least one of the
+elements in `values` i.e., the claim should be treated the same as if it did not
+exist in the Credential. Implementing this restriction may not be possible in
+all cases, for example, if the Wallet does not have access to the claim value
+before presentation or user consent or if another component routing
+the request to the Wallet does not have access to the claim value. Therefore, Verifiers
+must treat restrictions expressed using `values` as a best-effort way to improve
+user privacy, but MUST NOT rely on it for security checks.
+
+The purpose of the `claim_sets` syntax is to provide a way for a 
+verifier to describe alternative ways a given credential can satisfy the request. 
+The array ordering expresses the Verifier's preference for how the request must 
+be fulfilled. The first element in the array is the most preferred and the last 
+element in the array is the least preferred. Verifiers SHOULD use the principle of 
+least information disclosure to influence how they order these options. For example, a 
+proof of age request should prioritize requesting an attribute like `age_over_18` over
+an attribute like `birth_date`.
+
+If the Wallet cannot deliver all claims requested by the Verifier
+according to these rules, it MUST NOT return the respective Credential.
+
+#### Selecting Credentials
+
+The following rules apply for selecting Credentials via `credentials` and `credential_sets`:
+
+- If `credential_sets` is not provided, the Verifier requests all
+  Credentials in `credentials` to be returned.
+- Otherwise, the Verifier requests all of the credential set queries in the `credential_sets` array
+  with `required` evaluated as true to be returned at a minimum and optionally any of the credential set queries 
+  with `required` evaluated as false.
+- For each credential set query inside the `credential_sets` array, in order to
+  satisfy the query, the Wallet MUST return a credential or credentials that
+  match to one of the `options` inside the object. It is typically
+  expected that the Wallet presents the End-User with a choice of which
+  credential to present if multiple credentials match the query.
+
+Credentials not matching the respective constraints expressed within
+`credentials` MUST NOT be returned, i.e., they are treated as if
+they would not exist in the Wallet.
+
+If the Wallet cannot deliver all non-optional Credentials requested by the
+Verifier according to these rules, it MUST NOT return any credential(s).
+
+## Format-specific Properties {#format_specific_properties}
+
+The following format-specific properties are defined:
+
+### IETF SD-JWT VC {#format_vc_sd_jwt}
```

Comment:
I generally see risk in statically defining formats this query language works against. I would rather see us leverage a registry with associated guidance on how to plug into the general query language structure.

Reply:
similarly I would not support merging this PR without including at least w3c verifiable credentials - probably plain ID tokens/JWTs would be good too. the world is bigger than mdoc and ietf sd jwt vc

Reply:
this section is literally part of `## Format-specific Properties {#format_specific_properties}`, I made a suggestion to clarify that oid4vp is credential format agnostic and that "Other specifications or deployments can define their own Credential Format Profiles". Having said that, we can add w3c vdm profile of the query language, please open an issue and make a proposal. but that should not stop merging this PR as a starting point.

Reply:
https://github.com/openid/OpenID4VP/issues/284